scripts/touchpoint_api.py

```diff
@@ -0,0 +1,106 @@
+from argparse import ArgumentParser
+import requests
+import json
+import re
+
+# This script calls the touchpoint API to retrieve survey results for the provided document (form_id)
+# You will need to use your own Touchpoint API key:
+# touchpoint.py <form_id> <api_key> <start_date> <end_date>
+# if you want to save to the default file (touchpoint.csv), use the -p flag.
+
+
+
+MAX_PAGES = 500 # this can be changed to throttle if needed
+CSV_HEADER = 'id, date, referer, user agent, page, raw answer, language, english answer\n'
+
+def parse_args():
+    parser = ArgumentParser(
+        fromfile_prefix_chars="@", prog='Touchpoint Parser',
+        description='Parse specific results from a Touchpoint form',
+        epilog='Now, be off and about your business!', conflict_handler='error', add_help=True
+    )
+    parser.add_argument("form_id")
+    parser.add_argument("apikey")
+    parser.add_argument("start_date")
+    parser.add_argument("end_date")
+    parser.add_argument("filename")
+    parser.add_argument("-s", "--show", action="store_true") # to print results to console
+    parser.add_argument("-p", "--persist", action="store_true") # to save the results to the filename provided
+    return parser.parse_args()
+
+def make_single_call(base_url, page_number, start_date, end_date, api_key):
+    try:
+        response = requests.get(
+            f'{base_url}{page_number}&start_date={start_date}&end_date={end_date}',
+            headers={'X-Api-Key': api_key}
+        )
+        response.raise_for_status()
+        return response.json()
+    except requests.exceptions.RequestException as e:
+        print(f"Error making API request: {e}")
+        return None
+
+def strip_language(source):
+    if '/es/' in source:
+        return 'es'
+    elif '/fr/' in source:
+        return 'fr'
+    else:
+        return 'en'
```

Comment:
I'd worry about adding another place where we hardcode the site's languages, if/when we add more in the future. Is there a way we can extract this without explicitly listing them? e.g. reading from `_config.yml`'s existing `languages` configuration, or picking the first two letters of the path (probably not very reliable), or reading the `<html lang="xx">` value from the HTML itself.

Reply:
Understandable. The overall idea of this script was transitory and as a stop-gap measure until some display issues could be resolved. Getting the value from HTML wouldn't be possible, as this is taken from 'page' in the API response. However, adding it to a config file is a good way to go, since I'll need to remove the URL from the code as well (it fails the external links test).